ngraph/core/src/op/interpolate.cpp

```diff
@@ -111,143 +114,537 @@ namespace ngraph
         visitor.on_attribute("pads_end", m_ref.pads_end);
         return true;
     }
-}
+} // namespace ngraph
 
-// Interpolate v3
+// Interpolate v4
 
-constexpr NodeTypeInfo op::v3::Interpolate::type_info;
+constexpr NodeTypeInfo op::v4::Interpolate::type_info;
 
-op::v3::Interpolate::Interpolate(const Output<Node>& image,
+op::v4::Interpolate::Interpolate(const Output<Node>& image,
                                  const Output<Node>& output_shape,
-                                 const op::v3::Interpolate::InterpolateAttrs& attrs)
-    : Op({image, output_shape})
+                                 const Output<Node>& scales,
+                                 const Output<Node>& axes,
+                                 const op::v4::Interpolate::InterpolateAttrs& attrs)
+    : Op({image, output_shape, scales, axes})
+    , m_attrs(attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v4::Interpolate::Interpolate(const Output<Node>& image,
+                                 const Output<Node>& output_shape,
+                                 const Output<Node>& scales,
+                                 const op::v4::Interpolate::InterpolateAttrs& attrs)
+    : Op({image, output_shape, scales})
     , m_attrs(attrs)
 {
     constructor_validate_and_infer_types();
 }
 
-bool op::v3::Interpolate::visit_attributes(AttributeVisitor& visitor)
+bool op::v4::Interpolate::visit_attributes(AttributeVisitor& visitor)
 {
     visitor.on_attribute("attrs", m_attrs);
     return true;
 }
 
-void op::v3::Interpolate::validate_and_infer_types()
+static const char* cannot_define_axes =
+    "Cannot define axes of interpolation when there is "
```

Comment:
Could you please change to the following?
"Could not define axes of interpolation because there
are only two inputs and input data has a dynamic rank."

Reply:
Done.

Reply:
Check this one again please

Reply:
There is a typo in this file line 152: should be 'three' instead of 'two'.

Reply:
Fixed.